src/search.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+// Load external modules
+
+// Load internal modules
+const CloudSearchStrategy = require('./strategies/cloudsearch');
+const StubStrategy = require('./strategies/stub');
+
+class Search {
+  constructor(options, simulate) {
+    this._settings = options;
+    this._strategy = simulate ? new StubStrategy(options) : new CloudSearchStrategy(options);
+  }
+
+  uploadDocuments(searchContent) {
```

Comment:
If this is all the code does, it's enough to just use `return this._strategy..uploadDocument(this._settings, params)` instead of creating a new promise.